### PR TITLE
fix(core-android): Tabs selected item color gets applied to wrong Tabs or not applied at all

### LINKED ISSUE #8543: Tab label is not changing its color when swipe isdisabled
When swiping to another tab label is highlighted. When swipe is disabled label is NOT highlighted

https://play.nativescript.org/?template=play-ng&id=WxSW9I

this is very annoying issue because i cannot use tabs with RadViewList swipe actions :(
Can you fix it or provide workaround. I have tried to add selectedIndexChange event and assign it to property but this does not help either.

### PR DESCRIPTION
**Pull request by DimitrisRK**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/DevelopmentWorkflow.md#running-unit-tests.
- [x] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/WritingUnitTests.md.

## What is the current behavior?
There are couple of issues regarding Tabs navigation UX for some time now.
1) TabStrip selected color is not applied to the right tabstrip item if one changes tabs using swipe.
2) TabStrip selected color is not applied to any tabstrip items after tab change, if swipe is disabled.
3) Property `isAnimationEnabled` is not working for Tabs with swipeEnabled set to false.

## What is the new behavior?
This PR ensures colors are properly changed in all cases and animation to be configurable in all cases.
In terms of performance, this method won't try to update colors when not needed.
Additionally, it prevents the pointless calls of `setCurrentItem` every time `selectedIndexProperty` changes.

Until now, the following performance issue had been occuring:
`setCurrentItem` calls a native method which by its turn calls `setPrimaryItem` method.
`setPrimaryItem` is overridden inside Tabs UI component and changes `selectedIndex` property.
Upon change, `selectedIndex` property will call `setCurrentItem` again (called using `setNative` symbol)

Luckily, since `setCurrentItem` tried to set the same index there was no recursion but still executed tasks that shouldn't.

I'm positive this is a very good solution to get rid of some `TabStrip` UX bugs.
People are desperate with Tab components and that may give a small relief for Android platform at least.

There are lots of Tabs examples in UI app where this can be tested, but I also made sure to add selected/unselected colors to swipe-disabled-page and tabs-position-page examples.

Hope you consider this as a good alternative.

Fixes/Closes #8543 #8551 #8552 #8712

### CODE REVIEW

File: packages/core/ui/tabs/index.android.ts [hunk 1]
```diff
@@ -952,11 +952,9 @@ export class Tabs extends TabsBase {
 	}
 
 	[selectedIndexProperty.setNative](value: number) {
-		// TODO
-		// if (Trace.isEnabled()) {
-		//     Trace.write("TabView this._viewPager.setCurrentItem(" + value + ", " + smoothScroll + ");", traceCategory);
-		// }
-		this._viewPager.setCurrentItem(value, this.animationEnabled);
+		if (this._viewPager.getCurrentItem() != value) {
+			this._viewPager.setCurrentItem(value);
```
Comment (rigor789) [inline packages/core/ui/tabs/index.android.ts:956]: Shouldn't this pass the `animationEnabled` boolean?

**Review by rigor789 — COMMENTED** [review]
Not sure if intentional but `packages/core/platforms/android/widgets-release.aar` has been removed, I believe it just needs to be rebuilt with the native changes.

File: packages/core/ui/tabs/index.android.ts (hunk 1 above)
Comment (DimitrisRK) [reply packages/core/ui/tabs/index.android.ts:956]: @rigor789 This is handled in ui-mobile-base android class:
https://github.com/DimitrisRK/NativeScript/blob/tabs-selected/packages/ui-mobile-base/android/widgets/src/main/java/org/nativescript/widgets/TabViewPager.java#L58

Note: This java code is not an addition of mine. `animationEnabled` check had always been inside overridden method but it had an extra `swipeEnabled` check which was breaking proper functionality for all cases.

**Comment by DimitrisRK** [comment]
> Not sure if intentional but `packages/core/platforms/android/widgets-release.aar` has been removed, I believe it just needs to be rebuilt with the native changes.

Sorry, just noticed first comment. It's not removed, just looks like this because it's modified (few lines were left unused, so they got removed).
I modified and rebuilt it as expected.

**Review by rigor789 — APPROVED** [review]

**Pull request merged**